src/core/lib/surface/call.cc

```diff
@@ -1101,6 +1101,7 @@ static grpc_stream_compression_algorithm decode_stream_compression(
 static void publish_app_metadata(grpc_call* call, grpc_metadata_batch* b,
                                  int is_trailing) {
   if (b->list.count == 0) return;
+  if (is_trailing && call->buffered_metadata[1] == nullptr) return;
```

Comment:
It'd be better to find a fix in chttp2: fixes at this layer are more likely to break with efforts to remove this layer entirely in the c++ stack.

Reply:
Right, this is a temporary bandaid. We want to actually return a 400 from within chttp2, but "stop the bleeding first".